### DISCUSSION TITLE
Project Website from a single qmd file

### DISCUSSION THREAD

**Discussion by bcdavasconcelos** [quarto-dev/quarto-cli#4236] [Q&A]
Sometimes, it can be easier having it all together. That is, I mean, the whole book in a single `.qmd` file with all the Quarto and Pandoc configurations in the YAML front matter, then all the chapters, appendix and so on. Now, I can make that work reasonably well for creating a PDF Book output. I am curious about the following: could a website be created in the same way? I do not mean, of course, the html output of the Book project type which would keep the whole text in a single page. Nor I am concerned with other configurations which could be left for secondary auxiliary files (`css`, `js` and so on). I mean just the content of the `.qmd` files in one place, but with a functional left-sidebar for navigating the chapters. I see why that would be difficult, if not, impossible, without actually splitting up the original file, or implementing some other simple way of indicating a `{{page break}}` which would then also help prepare the sidebar content. 

I wonder if anything of the sort has been conceived or attempted with Quarto. Thank you for your time.

**Comment by mcanouil** [comment]
> Sometimes, it can be easier having it all together

I can't how it can makes things easier.


I am not sure to get your use case. Would you mind to elaborate what's seem to be hard in the fact to have `_quarto.yml` and Quarto document(s)?


Also note that for "one file", you should consider `format: html`.

**Comment by bcdavasconcelos** [comment]
In some writing applications, you end up with one compiled final draft. In this particular case, I am using [Scrivener](https://www.literatureandlatte.com/scrivener/overview) ([see also](https://forum.literatureandlatte.com/t/scrivener-quarto-a-technical-academic-publishing-workflow/129769?u=bernardo_vasconcelos)).  Having a separate `_quarto.yml` is fine: splitting up the `qmd` files, fixing all the internal links, and filling up the list of chapters (could be avoided with a `glob`, sure), is where the headache starts.

**Reply by mcanouil** [reply]
I am aware of Scrivener (it's also an item in the GitHub Awesome Quarto list) but I have to admit that I don't know much about it.

For me it seems to be a niche.
Maybe a post-render script could help do the heavy lifting (it was not a feature available at that time).

**Reply by bcdavasconcelos** [reply]
Thanks for getting back to me. I just remembered that Pandoc 3.0 now features the chunked HTML output! Maybe therein lies the solution.

**Reply by mcanouil** [reply]
If at some point you manage to have some kind of concrete example showing the need, I believe it can help Quarto team find a solution and suggest one that will suit your use case. 😉

About Pandoc 3.0, I don't know if it's going to be part of Quarto 1.3 release.

**Reply by cscheid** [reply]
> About Pandoc 3.0, I don't know if it's going to be part of Quarto 1.3 release.




yes, it will.
